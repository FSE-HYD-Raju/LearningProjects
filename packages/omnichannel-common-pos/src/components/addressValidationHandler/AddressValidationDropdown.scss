@import "../../styles/_variables.scss";

.AddressValidationDropdown {
	&-container {
		position: relative;
		left: 2.5rem;
		top: -2.8125;

		.OcDropdown {
			&-container {
				min-width: 18.75rem;
				left: 100%;
				right: 0;
				top: 100%;

				#address-validation-save-anyway-confirm-checkbox {
					margin-top: 1rem;
					margin-right: 0.35rem;
				}

				#addressSaveButton {
					justify-content: end;
				}
			}

			&-inner-container {
				flex-direction: row;

				&.OcDropdown-inner-container-left {
					margin-top: 0 !important;
				}
			}

			&-triangle {
				border-width: 0.625rem 0.625rem 0.625rem 0;
				border-color: transparent $babyblue transparent transparent;
				margin: 0.9375rem 0;
			}

			&-content {
				background: $babyblue;
				border-top: 0.1875rem solid $babyblue;
				padding: 0.3125rem 0.625rem 1rem 1.25rem;

				.address-not-found-message {
					font-weight: bold;
				}

				.OcInput-label {
					position: absolute;
				}
			}
		}

		&-proposals-container {
			padding: 0.9375rem;
		}
	}
}
