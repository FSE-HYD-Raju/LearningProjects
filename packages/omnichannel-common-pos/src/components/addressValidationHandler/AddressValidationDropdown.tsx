import { isEmpty } from "lodash";
import * as R from "react";
import onClickOutside, { HandleClickOutside, InjectedOnClickOutProps } from "react-onclickoutside";
import classnames from "classnames";
import { PostalAddress } from "../../redux/types";
import PostalAddressUtil from "../../utils/user/PostalAddressUtil";
import { FormattedMessage, FormattedHTMLMessage } from "../../channelUtils/";
import cssns from "../../utils/cssnsConfig";
import OcDropdown from "../ocComponents/OcDropdown";
import OcInput from "../ocComponents/OcInput";
import messages from "./AddressValidation.messages";
import AddressValidationHandlerProposals from "./AddressValidationHandlerProposals";
import { ContextType, contextTypesValidationMap } from "../../types";

const React = cssns("AddressValidationDropdown").React as typeof R;

interface AddressValidationDropdownStateProps {
	displayConfirmation: boolean;
	validationIsMandatory: boolean;
	proposals: PostalAddress[] | null;
	address: PostalAddress;
	proposalSelected?: PostalAddress;
	className: string;
}

interface AddressValidationDropdownActionProps {
	actions: {
		onChange: (address: PostalAddress) => void;
		getSaveButton: () => React.ReactElement<any> | null;
		forceSaveAddress: (isChecked: boolean) => void;
		cancel: () => void;
	};
}

interface AddressValidationDropdownState {
	isSaveInvalidAddressConfirmed: boolean;
}

type Props = AddressValidationDropdownStateProps & AddressValidationDropdownActionProps;

class AddressValidationDropdown extends (React as typeof R).Component<Props & InjectedOnClickOutProps, AddressValidationDropdownState>
	implements HandleClickOutside<any> {
	static contextTypes = contextTypesValidationMap;

	constructor(props: Props & InjectedOnClickOutProps, context: ContextType) {
		super(props, context);
		this.state = {
			isSaveInvalidAddressConfirmed: false
		};

		if (process.env.BROWSER) {
			props.enableOnClickOutside();
		}
	}

	handleClickOutside = () => {
		this.props.actions.cancel();
	};

	thereIsProposals = () => {
		return !isEmpty(this.props.proposals);
	};

	onConfirmSaveAnyway = (e: any) => {
		if (!e.target.checked) {
			this.props.actions.forceSaveAddress(false);
			this.setState({ isSaveInvalidAddressConfirmed: false });
			return;
		}

		this.setState({ isSaveInvalidAddressConfirmed: true });
		this.props.actions.forceSaveAddress(true);
	};

	render() {
		const addressParts = PostalAddressUtil.getAddressForMessage(this.props.address);
		const addressAsString = PostalAddressUtil.getAddressAsString(addressParts);

		const { formatMessage } = this.context.intl;

		const containerClasses = classnames({
			container: true,
			[this.props.className]: this.props.className
		});

		return (
			<div className={containerClasses}>
				<OcDropdown
					dropdownKey="AddressValidationDropdown-addresses"
					trianglePosition={this.props.className.includes("inner-block-message") ? "top" : ""}
				>
					{!this.thereIsProposals() &&
						this.props.validationIsMandatory && (
							<div className="form">
								<p className="address-not-found-message" data-test-name="address-not-found-message">
									<FormattedMessage
										{...messages.addressNotFound}
									/>
								</p>
								<FormattedHTMLMessage
									{...messages.addressNotFoundFor}
									values={{address: addressAsString}}
								/>
							</div>
						)}
					{(this.thereIsProposals()) && (
						<div className="proposals-container">
							<AddressValidationHandlerProposals
								proposals={this.props.proposals}
								proposalSelected={this.props.proposalSelected}
								currentAddress={this.props.address}
								validationIsMandatory={this.props.validationIsMandatory}
								confirmAnyway={this.state.isSaveInvalidAddressConfirmed}
								actions={{
									onChange: this.props.actions.onChange
								}}
							/>
						</div>
					)}
					{/* better store the 'checked' status to app state.. can't test this */}
					{this.props.displayConfirmation && <OcInput
						type="checkbox"
						name="address-validation-save-anyway-confirm-checkbox"
						onChange={this.onConfirmSaveAnyway}
						label={formatMessage(messages.myAddressIsNotOnTheListConfirmItIsValid)}
						checked={this.state.isSaveInvalidAddressConfirmed}
						standalone={true}
						id="address-validation-save-anyway-confirm-checkbox"
					/>}
					{this.props.actions.getSaveButton()}
				</OcDropdown>
			</div>
		);
	}
}

export default onClickOutside<Props>(AddressValidationDropdown);
