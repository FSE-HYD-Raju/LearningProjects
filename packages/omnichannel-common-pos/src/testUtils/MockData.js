import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
