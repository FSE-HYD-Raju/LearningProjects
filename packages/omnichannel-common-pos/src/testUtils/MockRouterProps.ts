import {
	RouteComponentProps,

} from "react-router";

import {
	Href,
	UnregisterCallback,
	TransitionPromptHook,
	LocationState,
	Path,
	LocationDescriptorObject,
	LocationListener
} from "history";

const mockLocation: RouteComponentProps<any>["location"] = {
		hash: "",
		key: "",
		pathname: "",
		search: "",
		state: {}
};

const mockHistory: RouteComponentProps<any>["history"] = {
	length: 2,
	action: "POP",
	location: mockLocation,
	push: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	replace: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	go: (n: number): void => {},
	goBack: (): void => {},
	goForward: (): void => {},
	block: (prompt?: boolean | string | TransitionPromptHook): UnregisterCallback => { return () => {}; },
	listen: (listener: LocationListener): UnregisterCallback => {return () => {}; },
	createHref: (location: LocationDescriptorObject): Href => { return ""; },
};

const mockRouterProps: RouteComponentProps<any> = {
		match: {
			isExact: true,
			params: {},
			path: "",
			url: ""
		},
		location: mockLocation,
		history: mockHistory,
		staticContext: {}
};

export {
	mockRouterProps,
	mockLocation,
	mockHistory,
	RouteComponentProps
};
