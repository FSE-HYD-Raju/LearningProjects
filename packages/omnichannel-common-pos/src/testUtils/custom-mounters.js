import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
