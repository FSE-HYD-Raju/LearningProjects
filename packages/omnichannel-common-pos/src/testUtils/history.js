import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
