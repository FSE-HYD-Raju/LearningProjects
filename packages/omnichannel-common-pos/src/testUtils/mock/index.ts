import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
