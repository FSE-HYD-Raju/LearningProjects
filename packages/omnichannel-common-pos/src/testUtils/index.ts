import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
