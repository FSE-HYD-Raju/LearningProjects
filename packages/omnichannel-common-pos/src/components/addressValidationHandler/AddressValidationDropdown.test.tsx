import * as React from "react";
import { shallowWithContext, mountWithContext, TestUtils } from "../../testUtils";
import AddressValidationDropdown from "./AddressValidationDropdown";

jest.mock("./AddressValidationHandlerProposals", () => {
    return {
        default: () => "MockAddressValidationHandlerProposals"
    };
});

describe("AddressValidationDropdown", () => {
	const proposals = [
		{
			street: "Mikonkatu 1",
			postalCode: "30500",
			city: "Mikkeli",
			country: "Finland"
		},
		{
			street: "Mikonkatu 1",
			postalCode: "60500",
			city: "Pori",
			country: "Finland"
		},
		{
			street: "Mikonkatu 1",
			postalCode: "10500",
			city: "Helsinki",
			country: "Finland"
		}
	];

	const minimumProps = {
        displayConfirmation: false,
        validationIsMandatory: false,
        proposals: null,
        address: {
            street: "Testikatu 1",
            postalCode: "00100",
            city: "Helsinki",
            country: "FI"
        },
        proposalSelected: undefined,
        className: "",
		actions: {
            onChange: jest.fn(),
            getSaveButton: jest.fn(),
            forceSaveAddress: jest.fn(),
            cancel: jest.fn(),
		}
	};

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(
			<AddressValidationDropdown {...minimumProps} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
        mountWithContext(<AddressValidationDropdown {...minimumProps} />);
	});

    it("renders 'address not found' message when there are no proposals and validation is mandatory", () => {
        const props = {
            ...minimumProps,
            validationIsMandatory: true
        };
        const wrapper = mountWithContext(<AddressValidationDropdown {...props} />);
        expect(wrapper.find("[data-test-name=\"address-not-found-message\"]")).toHaveLength(1);
    });

    it("renders proposals", () => {
        const props = {
            ...minimumProps,
            proposals
        };

        const wrapper = mountWithContext(<AddressValidationDropdown {...props} />);
        expect(wrapper.find("default")).toHaveLength(1); /* mocked AddressValidationHandlerProposals */
    });

    it("renders checkbox for confirming entered address to be valid, and no proposals presented", () => {
        const props = {
            ...minimumProps,
            displayConfirmation: true
        };

        const wrapper = mountWithContext(<AddressValidationDropdown {...props} />);
        expect(wrapper.find("input#address-validation-save-anyway-confirm-checkbox")).toHaveLength(1);
    });

    /* check comment in component code next to the input */
    xit("clicking on 'confirm address to be valid' checkbox gets given forceSaveAddress() called", () => {
        const props = {
            ...minimumProps,
            displayConfirmation: true
        };
        let wrapper = mountWithContext(<AddressValidationDropdown {...props} />);
        // console.log(wrapper.debug());

        console.log(wrapper.find("input#address-validation-save-anyway-confirm-checkbox").debug());
        wrapper = wrapper.setState({isSaveInvalidAddressConfirmed: true});
        // wrapper.find("input#address-validation-save-anyway-confirm-checkbox").simulate("click");
        wrapper.find("input#address-validation-save-anyway-confirm-checkbox").simulate("change", {target: {value: "1"}});
        expect(minimumProps.actions.forceSaveAddress).toHaveBeenCalled();
    });

    /* function is not called, don't know why .. */
    xit("calls given cancel() when clicked outside", () => {
        const wrapper = mountWithContext(<AddressValidationDropdown {...minimumProps} />);

        TestUtils.clickOnDocumentBody(() => {});
        expect(minimumProps.actions.cancel).toHaveBeenCalled();
    });
});
