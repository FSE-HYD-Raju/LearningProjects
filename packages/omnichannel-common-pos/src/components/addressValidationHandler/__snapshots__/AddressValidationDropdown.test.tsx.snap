// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressValidationDropdown succeeds at shallow mount with minimum props 1`] = `
<AddressValidationDropdown
  actions={
    Object {
      "cancel": [MockFunction],
      "forceSaveAddress": [MockFunction],
      "getSaveButton": [MockFunction],
      "onChange": [MockFunction],
    }
  }
  address={
    Object {
      "city": "Helsinki",
      "country": "FI",
      "postalCode": "00100",
      "street": "Testikatu 1",
    }
  }
  className=""
  disableOnClickOutside={[Function]}
  displayConfirmation={false}
  enableOnClickOutside={[Function]}
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  proposals={null}
  stopPropagation={false}
  validationIsMandatory={false}
/>
`;
